src/main/java/com/celements/job/AbstractJob.java

```diff
@@ -0,0 +1,172 @@
+import org.xwiki.context.ExecutionContextManager;
+
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.plugin.scheduler.XWikiServletRequestStub;
+import com.xpn.xwiki.plugin.scheduler.XWikiServletResponseStub;
+import com.xpn.xwiki.web.Utils;
+import com.xpn.xwiki.web.XWikiResponse;
+import com.xpn.xwiki.web.XWikiServletRequest;
+
+/**
+ * Base class for any XWiki Quartz Job. This class take care of initializing ExecutionContext
+ * properly.
+ * <p>
+ * A class extending {@link AbstractJob} should implements {@link #executeJob(JobExecutionContext)}.
+ *
+ * @since 2.90
+ * @author fabian pichler
+ */
+public abstract class AbstractJob implements Job {
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.quartz.Job#execute(org.quartz.JobExecutionContext)
+   */
+  @Override
+  public final void execute(JobExecutionContext jobContext) throws JobExecutionException {
+    JobDataMap data = jobContext.getJobDetail().getJobDataMap();
+
+    // The XWiki context was saved in the Job execution data map. Get it as we'll retrieve
+    // the script to execute from it.
+    XWikiContext xwikiContext = (XWikiContext) data.get("context");
+
+    Execution execution;
+    try {
+      execution = initExecutionContext(xwikiContext);
+    } catch (ExecutionContextException | DocumentNotExistsException | MalformedURLException exp) {
+      throw new JobExecutionException("Fail to initialize execution context", exp);
+    }
+
+    try {
+      // Execute the job
+      executeJob(jobContext);
+    } catch (Throwable t) {
+      getLogger().error("Exception thrown during job execution", t);
+    } finally {
+      // We must ensure we clean the ThreadLocal variables located in the Execution
+      // component as otherwise we will have a potential memory leak.
+      execution.removeContext();
+    }
+  }
+
+  private Execution initExecutionContext(XWikiContext xwikiContext)
+      throws ExecutionContextException, DocumentNotExistsException, MalformedURLException {
+    // Init execution context
+    ExecutionContextManager ecim = Utils.getComponent(ExecutionContextManager.class);
+    Execution execution = Utils.getComponent(Execution.class);
+
+    ExecutionContext ec = new ExecutionContext();
+    // Bridge with old XWiki Context, required for old code.
+    ec.setProperty("xwikicontext", createJobContext(xwikiContext));
+
+    ecim.initialize(ec);
+    execution.setContext(ec);
+    return execution;
+  }
+
+  /**
+   * create a new copy of the xwiki context. Job Executions always use a different thread.
+   * The xwiki context is NOT thread safe and may not be shared.
+   *
+   * @throws DocumentNotExistsException
+   * @throws MalformedURLException
+   */
+  XWikiContext createJobContext(XWikiContext xwikiContext) throws DocumentNotExistsException,
```

Comment:
please extract a `cloneXWikiContext` method to another class for reusability, we will need it again (see issue)